track/track.py

```diff
@@ -0,0 +1,250 @@
+        self.start_frame += start
+        self.bounds_history = self.bounds_history[start:end-1]
+
+    def get_track_region_score(self, region: Region):
+        """
+        Calculates a score between this track and a region of interest.  Regions that are close the the expected
+        location for this track are given high scores, as are regions of a similar size.
+        """
+        expected_x = int(self.bounds.mid_x + self.vel_x)
+        expected_y = int(self.bounds.mid_y + self.vel_y)
+
+        distance = ((region.mid_x - expected_x) ** 2 + (region.mid_y - expected_y) ** 2) ** 0.5
+
+        # ratio of 1.0 = 20 points, ratio of 2.0 = 10 points, ratio of 3.0 = 0 points.
+        # area is padded with 50 pixels so small regions don't change too much
+        size_difference = (abs(region.area - self.bounds.area) / (self.bounds.area+50)) * 100
+
+        return distance, size_difference
+
+    def get_overlap_ratio(self, other_track, threshold = 0.05):
+        """
+        Checks what ratio of the time these two tracks overlap.
+        :param other_track: the other track to compare with
+        :param threshold: how much frames must be overlapping to be counted
+        :return: the ratio of frames that overlap
+        """
+
+        if len(self) == 0 or len(other_track) == 0:
+            return 0.0
+
+        start = max(self.start_frame, other_track.start_frame)
+        end = min(self.start_frame + len(self), other_track.start_frame + len(other_track))
+
+        frames_overlapped = 0
+
+        for pos in range(start, end+1):
+            our_index = pos - self.start_frame
+            other_index = pos - other_track.start_frame
+            if our_index >= 0 and other_index >= 0 and our_index < len(self) and other_index < len(other_track):
+                our_bounds = self.bounds_history[our_index]
+                other_bounds = other_track.bounds_history[other_index]
+                overlap = our_bounds.overlap_area(other_bounds) / our_bounds.area
+                if overlap >= threshold:
+                    frames_overlapped += 1
+
+        return frames_overlapped / len(self)
+
+    @property
+    def mass(self):
+        return self.bounds_history[-1].mass
+
+    @property
+    def bounds(self) -> Region:
+        return self.bounds_history[-1]
+
+    @property
+    def duration(self):
+        """ Duration of track in seconds. """
+        return len(self) / 9.0
+
+    @property
+    def end_time(self):
+        return self.start_time + datetime.timedelta(seconds=self.duration)
+
+    def __repr__(self):
+        return "Track:{} frames".format(len(self))
+
+    def __len__(self):
+        return len(self.bounds_history)
+
+TrackMovementStatistics = namedtuple(
+    'TrackMovementStatistics',
+    'movement max_offset score average_mass median_mass duration delta_std'
+)
+TrackMovementStatistics.__new__.__defaults__ = (0,) * len(TrackMovementStatistics._fields)
```

Comment:
Ick. We should use the attrs package instead of nametuples. Much cleaner. Don't worry about this for now though.